File: 프로그래머스/연습문제/Lv1/콜라츠 추측/콜라츠 추측.py
def solution(num):
    if num == 1: return 0

    answer = 0
    for i in range(500):
        answer += 1
        if num%2 == 0: num //= 2
        else: num = num*3+1

        if num == 1: return answer

    return -1

File: 프로그래머스/연습문제/Lv1/콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    if(num == 1) return 0;
    
    int answer = 0;
    long long n = num;
    for(int i=0; i<500; i++){
        answer++;
        if(!(n%2)) n /= 2;
        else n = n*3+1;
        
        if(n == 1) return answer;
    }
    
    return -1;
}
